### PR TITLE
fix: hygiene bundle — verify-plugin schema count + idea_summary cap + macOS dup gitignore (#63 #65 #68)

### LINKED ISSUE #63: 🟡 I-1: verify-plugin.sh assertion broken — schema count 6 vs hardcoded 5
**Priority**: 🟡 Important
**Audit source**: T4 Leader (devops-architect)
**Related PR**: v1.11.0 (PR #54 or earlier schema addition)

## Problem

Running `scripts/verify-plugin.sh` on main is **currently failing**.

### Evidence

```bash
$ bash scripts/verify-plugin.sh
...
✗ schemas: 6 (expected 5)
...
=== SUMMARY ===
Pass: 56
Fail: 1
```

**File**: near `scripts/verify-plugin.sh:131` — the schema count assertion is hardcoded to `-eq 5`, but v1.11.0 added a 6th schema.

### Side effects

- DoD promised the pass count would grow from 56 → 70+
- Reality: still 56, plus 1 broken assertion
- That is, the v1.11.0 PR was merged while ignoring verify-plugin verification (CI gating likely does not mark this script as required)

## Acceptance Criteria

- [ ] Fix the assertion near `scripts/verify-plugin.sh:131`: use `expected_schemas=6` or a dynamic count
- [ ] Add a helper function so hardcoded assertion counts auto-sync:
  ```bash
  count_files() { find "$1" -name "$2" | wc -l | tr -d ' '; }
  EXPECTED_SCHEMAS=$(count_files "$PLUGIN_DIR/schemas" "*.schema.json")
  # ... assertions should only check the invariant ("schema count has not drifted abnormally")
  ```
- [ ] Confirm `.github/workflows/ci.yml` gates on the verify-plugin result with fail-on-error (add it if not already enforced)
- [ ] Audit the DoD pass count 70+ promise — confirm whether it grew naturally from new fixtures/assets, then update the expected number

## Notes

A trivial fix (1-line change), but the real concern is that main is partially broken — CI gating policy needs review.

## Cross-references

- T-1 (advocate count assertion) — same pattern
- C-1 (missing e2e verification) — the assertions themselves are stale

### LINKED ISSUE #65: 🟡 I-3: idea_summary uncapped maxLength — DoS via prompt expansion
**Priority**: 🟡 Important
**Audit source**: T2 Leader (security-engineer) — strongest residual risk
**Related file**: `plugins/preview-forge/schemas/idea-spec.schema.json:9-13`

## Problem

The S-3 fix applied maxLength/maxItems to `must_have_constraints` / `non_goals`, but the **`idea_summary` field has only minLength:1 and no maxLength**.

### Evidence

```json
// plugins/preview-forge/schemas/idea-spec.schema.json:9-13
"idea_summary": {
  "type": "string",
  "minLength": 1,
  "description": "..."
}
```

### Attack scenario

A malicious seed or cache replay injects a 10MB `idea_summary` →
1. Schema validation passes (only minLength)
2. The value gets inlined into 26 advocate prompts → token expansion explodes
3. `_filled_ratio` always counts idea_summary as filled (1), so the fallback never triggers
4. The cache-key sha256 also has to chew through the 10MB input

## Implications

The S-3 mitigation loses internal consistency. Capping every other string field but leaving idea_summary unbounded is inconsistent design.

## Acceptance Criteria

- [ ] Add `"maxLength": 5000` at `plugins/preview-forge/schemas/idea-spec.schema.json:9-13`:
  ```json
  "idea_summary": {
    "type": "string",
    "minLength": 1,
    "maxLength": 5000,
    "description": "..."
  }
  ```
  (5000 is a conservative cap that comfortably accommodates a 1-line idea + a few sentences while preventing abuse)
- [ ] Add CI fixture `tests/fixtures/security/oversized-idea-summary.json`
  - Assert that a 5001-char idea_summary fails schema validation
- [ ] Add the fixture verification step to `tests/fixtures/security/verify-security.sh`

## Test scenario

```python
import json, jsonschema

schema = json.load(open("plugins/preview-forge/schemas/idea-spec.schema.json"))

# Should pass:
short = {"idea_summary": "x" * 100, "_schema_version": "1.0", ...}
jsonschema.validate(short, schema)

# Should fail:
huge = {"idea_summary": "x" * 5001, ...}
try: jsonschema.validate(huge, schema); assert False
except jsonschema.ValidationError: pass  # expected
```

## Cross-references

- S-3 (must_have_constraints maxLength) — already shipped, same pattern
- LESSON 0.4 (zero-dep) — jsonschema is already a hard dep
- C-1 (e2e run validating large idea_summary inputs)

## Notes

A 1-line schema change plus 1 fixture file. Highest-leverage risk reduction.

### LINKED ISSUE #68: 🟡 I-6: idea-spec.schema 2.json macOS duplicate file — workspace pollution
**Priority**: 🟡 Important
**Audit source**: T5 Expert (technical-writer) — "most jarring inconsistency"
**Related file**: `plugins/preview-forge/schemas/idea-spec.schema 2.json` (deletion target)

## Problem

The `plugins/preview-forge/schemas/` directory contains a **duplicate file created by macOS Finder that is committed and present on main**.

### Evidence

```bash
$ ls plugins/preview-forge/schemas/
idea-spec.schema.json
idea-spec.schema 2.json  ← STALE COPY
panel-vote.schema.json
pf-profile.schema.json
preview-card.schema.json
score-report.schema.json
```

Contents of `idea-spec.schema 2.json`:
- Pre-v1.10 description (e.g. "If < 0.5, low_spec_quality" message)
- No `maxItems: 20`
- No `enum` constraint
- Legacy `_filled_ratio` rule

The canonical file (`idea-spec.schema.json`) is the one referenced by all docs and code. The duplicate is **never referenced but sits on disk**.

## Implications

1. **Workspace hygiene**: explicit violation of RULES.md "no artifact pollution"
2. **CI risk**: tools that iterate the schemas directory alphabetically could pick the stale schema
3. **Judge optics**: a reviewer browsing the schema folder sees two near-identical files → suspects "drift in progress?"
4. **Affects verify-plugin.sh schema count**: directly tied to I-1 — this is one reason the count is 6 vs expected 5

## Acceptance Criteria

- [ ] Plain delete: `git rm "plugins/preview-forge/schemas/idea-spec.schema 2.json"`
- [ ] Add macOS duplicate patterns to `.gitignore`:
  ```gitignore
  # macOS Finder/iCloud duplicates
  *\ [0-9].*
  *\ copy.*
  *\ copy\ [0-9].*
  ```
- [ ] After the PR merges, confirm the `verify-plugin.sh` schema count matches expected (reinforces I-1)
- [ ] Sweep other schemas directories for similar duplicates:
  ```bash
  find plugins/preview-forge -name "* [0-9].*" -o -name "*copy*"
  ```

## Cross-references

- I-1 (verify-plugin.sh schema count broken) — partially resolved by closing this issue
- RULES.md "Workspace Hygiene" rule

## Notes

`git mv` or a plain `git rm`. One command. Risk 0.

### PR DESCRIPTION
**Pull request by sgwannabe**
## Summary

- **#63 I-1**: `scripts/verify-plugin.sh` schema count was hardcoded `5`; replaced with dynamic count + floor 5 invariant so it auto-adjusts when W2.8 lands the planned 6th schema (`spec-anchor-audit`) and still catches accidental schema removal below the v1.6.0 baseline.
- **#65 I-3**: `schemas/idea-spec.schema.json` `idea_summary` capped at `maxLength: 5000` (was minLength-only). Prevents schema-valid 10MB poisoned idea_summary from inflating every advocate prompt past Rule 9 containment.
- **#68 I-6**: `.gitignore` blocks macOS Finder/iCloud duplicate patterns (`* [0-9].*`, `* copy`, `* copy [0-9].*`, etc.). The original duplicate schema file is already absent from the tree; this PR adds the structural guard.

## Test plan

- [x] `bash scripts/verify-plugin.sh` → Pass 57 / Fail 0
- [x] `bash tests/fixtures/security/verify-security.sh` → all Phase-1 defenses hold; new boundary cases (5001 chars rejected for `maxLength` on `idea_summary` specifically; 5000-char twin accepted) pass

## Codex review

Single pass: **0 P1**, 2 P2 noted as out-of-scope follow-ups (will track in a separate issue):
1. Reject/normalize `/pf:new` ideas >5000 chars on the input path (in `commands/new.md`) — input-path enforcement layer beyond schema validation.
2. Enforce the cap on the cache replay loader read path (beyond `verify-security.sh`) — defense in depth.

Both go beyond this PR's "schema + fixture only" minimal scope.

Closes #63
Closes #65
Closes #68

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -24,6 +24,23 @@ runs/
...
+# macOS Finder/iCloud duplicates (#68 — block patterns like
+# "idea-spec.schema 2.json", "README.md copy", "report copy 3.pdf"
+# that Finder/iCloud silently spawn on conflict).
+# Cover both single- and double-digit suffixes (Finder uses sequential
+# numbers; >9 conflicts is rare but possible — gitignore is cheap, the
+# 26-advocate scan that picks up `*.md` is not). The bare `* copy` /
+# `* copy N` forms (no extension) are emitted by Finder when copying
+# extension-less files like LICENSE.
+*\ [0-9].*
+*\ [0-9][0-9].*
```
[the remark this answers is not in the document]
Comment (sgwannabe) [reply .gitignore:38]: 좋은 지적 감사합니다. 확장자 없는 `LICENSE 2`, `NOTICE 12` 같은 Finder/iCloud 충돌 파일은 기존 `*\ [0-9].*` 패턴이 잡지 못하는 사각지대였습니다. 제안하신 4개 패턴(`*\ [0-9]`, `*\ [0-9].*`, `*\ [0-9][0-9]`, `*\ [0-9][0-9].*`)을 모두 추가했습니다. 커밋: 2a6bce7.

File: scripts/verify-plugin.sh [hunk 2]
```diff
@@ -127,8 +127,14 @@ seed_count=$(find "$PLUGIN_DIR/seed-ideas" -maxdepth 1 -name "[0-9][0-9]-*.md" |
...
 schema_count=$(find "$PLUGIN_DIR/schemas" -name "*.json" | wc -l | tr -d ' ')
-# v1.6.0: 5 schemas (preview-card, panel-vote, score-report, pf-profile, idea-spec).
-[[ "$schema_count" -eq 5 ]] && ok "5 JSON schemas (preview-card, panel-vote, score-report, pf-profile, idea-spec)" || bad "schemas: $schema_count (expected 5)"
+# Dynamic count (#63): assert that every `*.schema.json` under schemas/ is
+# accounted for, instead of hard-coding `5`. This auto-adjusts when a 6th
+# schema lands (e.g. spec-anchor-audit per W2.8 plan) without churning this
+# script. We still require the canonical v1.6.0 set as a floor.
+expected_schemas=$(find "$PLUGIN_DIR/schemas" -name "*.schema.json" | wc -l | tr -d ' ')
+[[ "$schema_count" -eq "$expected_schemas" && "$expected_schemas" -ge 5 ]] && \
+  ok "$expected_schemas JSON schemas (dynamic count, floor 5: preview-card, panel-vote, score-report, pf-profile, idea-spec)" || \
```
[the remark this answers is not in the document]
Comment (sgwannabe) [reply scripts/verify-plugin.sh]: 지적해 주신 대로 수정했습니다. 동적 카운트로 전환한 의도가 5개 이름을 하드코딩하면 무력화되므로, 메시지에서 스키마 이름 목록을 제거하고 `dynamic count, floor 5`로 일반화했습니다. 향후 6번째 스키마(예: spec-anchor-audit)가 추가돼도 메시지가 misleading해지지 않습니다. 커밋: 2a6bce7.

File: tests/fixtures/security/verify-security.sh [hunk 3]
```diff
@@ -128,6 +128,35 @@ try:
     sys.exit(1)
 except jsonschema.ValidationError as e:
     print(f"  ✓ malicious-constraints.json — schema rejected ({e.validator} on {list(e.absolute_path)[:3]})")
+
+# #65 — idea_summary maxLength cap (5000). 5001-char payload must reject;
+# a 5000-char twin must validate (proves the cap is the boundary, not a
+# coincidental other-rule reject like additionalProperties).
+oversize = json.load(open("$FIXTURES_DIR/oversized-idea-summary.json"))
```
[the remark this answers is not in the document]
Comment (sgwannabe) [reply tests/fixtures/security/verify-security.sh]: 지적은 일반론으로 타당합니다(security-high 관행상 `python3 -c` 인라인 스크립트에 셸 변수를 보간하지 않는 것이 방어적 best practice). 다만 이번 PR에서는 다음 두 가지 이유로 **유보(deferred)**합니다.

1. **위험 surface 평가**: `$FIXTURES_DIR`는 `cd "$(dirname "$0")" && pwd`로 산출되어 인자/환경변수가 아닌 스크립트 자기 위치에서 파생됩니다. 외부 입력 경로가 없어 실질적 injection 표면은 없습니다. 따라서 high-priority라기보다 hardening grade의 제안으로 봅니다.

2. **PR 분리 원칙(scope guard)**: 동일 heredoc 블록이 [S-3] 섹션(라인 111–160)을 통째로 감싸고 있고, S-1/S-2/S-3 섹션은 **PR #85**가 소유합니다. 본 PR은 #65(idea_summary cap) 검증을 위해 oversized 체크 12줄만 추가한 상태이므로, heredoc 전체를 argv 전달 방식(`python3 - "$FIXTURES_DIR" "$REPO_ROOT" <<PY ... sys.argv[1] ...`)으로 재구성하면 PR #85 영역까지 침범합니다.

후속 조치로 별도 hygiene 이슈로 분리해 PR #85 머지 이후 일괄 적용하겠습니다(`$FIXTURES_DIR` + `$REPO_ROOT` 두 변수 모두 argv화). 본 PR에서는 그대로 두겠습니다.

[the remark this answers is not in the document]
Comment (sgwannabe) [reply tests/fixtures/security/verify-security.sh]: Applied in a5c5c6d — #85가 머지되어 S-3 heredoc 영역 충돌 위험이 사라졌습니다. #85가 S-1/I-7에 적용한 것과 동일한 패턴(argv-pass + with-open + single-quoted heredoc)을 S-3에도 적용했습니다. 실제 인젝션 risk는 nil이었으나 프로젝트 정책상 Python 소스에 변수 보간을 금지하는 invariant를 강제했습니다.
